### ISSUE TITLE
Exception: Project might be already disposed in CloudLibraryMavenProjectState

### ISSUE THREAD

**Issue by ivanporty** [GoogleCloudPlatform/cloud-code-intellij#2234]
java.lang.AssertionError at ComponentManagerImpl.lambda$throwAlreadyDisposed$1 : Already disposed: Project (Disposed) 
Stack trace:
```
java.lang.AssertionError: Already disposed: Project (Disposed)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.lambda$throwAlreadyDisposed$1(ComponentManagerImpl.java:247)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:43)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:931)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:47)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:43)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.throwAlreadyDisposed(ComponentManagerImpl.java:245)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:238)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getComponent(ComponentManagerImpl.java:148)
	at com.intellij.openapi.module.ModuleManager.getInstance(ModuleManager.java:44)
	at com.google.cloud.tools.intellij.cloudapis.maven.CloudLibraryMavenProjectState.syncManagedProjectLibraries(CloudLibraryMavenProjectState.java:106)
	at com.google.cloud.tools.intellij.cloudapis.maven.CloudLibraryMavenProjectState$1.projectsScheduled(CloudLibraryMavenProjectState.java:84)
	at org.jetbrains.idea.maven.project.MavenProjectsManager.fireProjectScheduled(MavenProjectsManager.java:1227)
	at org.jetbrains.idea.maven.project.MavenProjectsManager.access$1200(MavenProjectsManager.java:68)
	at org.jetbrains.idea.maven.project.MavenProjectsManager$5.projectsUpdated(MavenProjectsManager.java:436)
	at org.jetbrains.idea.maven.project.MavenProjectsTree.fireProjectsUpdated(MavenProjectsTree.java:1482)
	at org.jetbrains.idea.maven.project.MavenProjectsTree.access$300(MavenProjectsTree.java:59)
	at org.jetbrains.idea.maven.project.MavenProjectsTree$UpdateContext.fireUpdatedIfNecessary(MavenProjectsTree.java:1539)
	at org.jetbrains.idea.maven.project.MavenProjectsTree.update(MavenProjectsTree.java:478)
	at org.jetbrains.idea.maven.project.MavenProjectsTree.updateAll(MavenProjectsTree.java:431)
	at org.jetbrains.idea.maven.project.MavenProjectsProcessorReadingTask.perform(MavenProjectsProcessorReadingTask.java:60)
	at org.jetbrains.idea.maven.project.MavenProjectsProcessor.doProcessPendingTasks(MavenProjectsProcessor.java:132)
	at org.jetbrains.idea.maven.project.MavenProjectsProcessor.access$000(MavenProjectsProcessor.java:32)
	at org.jetbrains.idea.maven.project.MavenProjectsProcessor$2.run(MavenProjectsProcessor.java:107)
	at org.jetbrains.idea.maven.utils.MavenUtil.lambda$runInBackground$5(MavenUtil.java:446)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:305)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```
Exception class name:
java.lang.AssertionError
Exception message:

Product Version:
2018.1.6

**Comment by sageprice** [comment]
We got another report for this today (July 30, 2019) from a user running `19.7.1` on `IntelliJ IDEA 2019.2`.

Not sure how we got into this state, but if the project is disposed of maybe we could check for that and discard the `CloudLibraryMavenProjectState` component. I'm not sure what else we can do, dunno how we can recover when the project is gone.

**Comment by matthewmichihara** [comment]
On quick inspection, that class overrides projectOpened, but not
projectClosed so maybe there's some cleanup we can do there.

**Comment by peterlin741** [comment]
Another instance of the same error, although the stack trace starts to differ halfway through: https://github.com/GoogleCloudPlatform/cloud-code-intellij/issues/2718

**Comment by etanshaul** [comment]
another instance. intellij 2019.3.2,CC 20.1.3 (pretty old)

**Comment by peterlin741** [comment]
Fixed in `20.8.1`, Can reopen if we see this again
